### PR TITLE
Mixxx 2.1 updates for DJ console 4-Mx

### PR DESCRIPTION
**Pull request by JosepMaJAZ**
2.1 effect racks
2.1 softTakeover fix for setParameter.
xml mapping fixes
sampler mapping fixes
single jog mode hack

### CODE REVIEW

**Comment by Be-ing** [comment]
Thanks for submitting this. I won't take a detailed look until after 2.1 beta is released, but at first glance I see a lot of legacy code for 2.0. Mixxx 2.0 was already released with a mapping for this controller and this PR will be modifying it for 2.1, so I don't think there's any need to maintain backwards compatibility with 2.0.

**Comment by Be-ing** [comment]
I will get to reviewing this soon, but for now I will be taking a break from development and code review for a few days.

File: res/controllers/Hercules-DJ-Console-4-Mx-scripts.js [hunk 1]
```diff
@@ -93,7 +95,10 @@ Hercules4Mx.userSettings = {
     // mixxx20 : As they were initially defined with the release of Mixxx 2.0
     // mixxx21 : AS they were defined with the release of Mixxx 2.1
     // With a little caution, you can modify them at the bottom of this file
-    'FXbuttonsSetup': 'mixxx21'
+    'FXbuttonsSetup': 'mixxx21',
+    // This allows to use the left jog wheel in place of the right jog wheel when pressing the right jog wheel.
+    // I implemented this to overcome the failure of the right jog wheel on my controller.
+    'useSingleJogWeelHack': false
```
Comment (Be-ing) [inline res/controllers/Hercules-DJ-Console-4-Mx-scripts.js]: typo: useSingleJogW**h**eelHack

File: res/controllers/Hercules-DJ-Console-4-Mx-scripts.js [hunk 2]
```diff
@@ -1620,6 +1648,8 @@ Hercules4Mx.getNewDestinationChannel = function(chan) {
 // FX button presets configuration.
 Hercules4Mx.reinitButtonsMap = function() {
     Hercules4Mx.buttonMappings = Hercules4Mx.buttonMappings.slice(Hercules4Mx.buttonMappings.length);
+    //This is setup here because the function is undefined when the map is defined.
+    Hercules4Mx.noActionButtonMap.buttonPressAction=Hercules4Mx.FxActionNoOp;
```
Comment (Be-ing) [inline res/controllers/Hercules-DJ-Console-4-Mx-scripts.js]: spaces around `=` operator

**Review by Be-ing — CHANGES_REQUESTED** [review]
Thank you for updating this and sorry for the delay reviewing it. I noted just a few minor nitpicks in the code. Could you remove the legacy code for 2.0? I don't think there's any point keeping it in 2.1.

**Comment by Be-ing** [comment]
Also, can you rebase this on the 2.1 branch?

**Comment by JosepMaJAZ** [comment]
Glups.... I believe I made a mistake with the rebase... Can you simply get the latest changes and commit them as they should?

**Comment by Be-ing** [comment]
I rebased this branch and directly pushed it to 2.1. Thank you @JosepMaJAZ.

**Comment by Be-ing** [comment]
Could you checkout the 2.1 branch and confirm that I didn't mess anything up with the rebasing?

**Comment by JosepMaJAZ** [comment]
I can confirm that the files have the expected changes.

**Pull request closed without merging**